client/lib/wp/sync-handler/index.js

```diff
@@ -0,0 +1,165 @@
+
+/**
+ * Module dependencies
+ */
+import config from 'config';
+import localforage from 'localforage';
+import Hashes from 'jshashes';
+
+import debugFactory from 'debug';
+const debug = debugFactory( 'calypso:sync-handler' );
+
+// defaults
+const defaults = config( 'sync-handler-defaults' );
+defaults.driver = localforage[ defaults.driver ];
+
+/**
+ * SyncHandler class
+ */
+export class SyncHandler {
+	/**
+	 * Create a SyncHandler instance
+	 *
+	 * @param {Object} [setup] - sync setup
+	 * @param {Function} handler - wpcom handler
+	 *
+	 * @return {Function} sync-handler wrapper
+	 */
+	constructor( setup, handler ) {
+		if ( 'function' === typeof setup ) {
```

Comment:
well ... https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#yoda-conditions
maybe I should swap them.